runtime/lua/vim/lsp/util.lua

```diff
@@ -822,43 +926,30 @@ function M.convert_signature_help_to_markdown_lines(signature_help)
   if not signature then
     return
   end
-  vim.list_extend(contents, vim.split(signature.label, '\n', true))
-  if signature.documentation then
-    M.convert_input_to_markdown_lines(signature.documentation, contents)
+
+  local active_parameter = signature.activeParameter or signature_help.activeParameter or 0
+  local label, active_offset = signature_label(signature, active_parameter, parametersOnly)
+
+  local hl_param = "LspSignatureParamActive"
+
+  table.insert(contents, label)
+  table.insert(highlights, { line = 0, col = 0, opts = { end_line = 0, end_col = label:len(), hl_group = "LspSignatureActive", priority = 200 } })
+  if active_offset then
+    table.insert(highlights, { line = 0, col = active_offset[1], opts = { end_line = 0, end_col = active_offset[2], hl_group = hl_param, priority = 100 } })
   end
-  if signature.parameters and #signature.parameters > 0 then
-    local active_parameter = signature_help.activeParameter or 0
-    -- If the activeParameter is not inside the valid range, then clip it.
-    if active_parameter >= #signature.parameters then
-      active_parameter = 0
-    end
-    local parameter = signature.parameters[active_parameter + 1]
-    if parameter then
-      --[=[
-      --Represents a parameter of a callable-signature. A parameter can
-      --have a label and a doc-comment.
-      interface ParameterInformation {
-        --The label of this parameter information.
-        --
-        --Either a string or an inclusive start and exclusive end offsets within its containing
-        --signature label. (see SignatureInformation.label). The offsets are based on a UTF-16
-        --string representation as `Position` and `Range` does.
-        --
-        --*Note*: a label of type string should be a substring of its containing signature label.
-        --Its intended use case is to highlight the parameter label part in the `SignatureInformation.label`.
-        label: string | [number, number];
-        --The human-readable doc-comment of this parameter. Will be shown
-        --in the UI but can be omitted.
-        documentation?: string | MarkupContent;
-      }
-      --]=]
-      -- TODO highlight parameter
-      if parameter.documentation then
-        M.convert_input_to_markdown_lines(parameter.documentation, contents)
+
+  for i = 1,#signature_help.signatures,1 do
```

Comment:
formatting (spaces), also is the last 1 necessary? you're iterating by one anyways

Reply:
I've no idea. Lua isn't a language I'm skilled at. Can I omit it?

Reply:
fixed